objcgen/NameGenerator.cs

```diff
@@ -120,5 +120,15 @@ public static string GetMonoName (Type t)
 				throw new NotImplementedException ($"Converting type {t.Name} to a mono type name");
 			}
 		}
+
+		public static string GetObjCParamName (Type pt, List<ProcessedType> allTypes)
```

Comment:
An example from the code below:
```
string ptname = string ptname = NameGenerator.GetObjCParamName (p.ParameterType, types);
  					objc.Append (":(").Append (ptname).Append (")").Append (p.Name); (p.ParameterType, types);
objc.Append (":(").Append (ptname).Append (")").Append (p.Name);
```
We call NameGenerator.GetObjCParamName on the parameter type, pass that intot he append as ptname, then immediately pass in p.Name as well.